api/routes/collections.js

```diff
@@ -119,56 +119,80 @@ router.get('/', validateCollectionSearchParams, async (req, res, next) => {
 
 /**
  * GET /collections/:id
- * Returns a single collection by ID. Includes all STAC Collection fields
- * (stac_version, type, title, description, license, extent, links, etc).
- * 
- * Returns:
- * - 200 OK with full Collection object if found
- * - 404 NotFound with proper error format if collection does not exist
+ * Returns a single collection by ID.
+ *
+ * Behaviour:
+ * - Uses the shared buildCollectionSearchQuery helper with an `id` filter
+ *   so that GET /collections and GET /collections/:id stay aligned.
+ * - Returns:
+ *   - 200 OK with a single Collection object if found
+ *   - 404 NotFound with standardized error body if the collection does not exist
+ *
+ * Note:
+ * - The exact shape / fields of the returned collection are controlled by the
+ *   SELECT part in buildCollectionSearchQuery. This allows the query builder
+ *   (and later a mapping layer) to evolve without touching this route.
  */
-router.get('/:id', (req, res) => {
-  // TODO: Create a proper validator middleware for :id parameter to avoid SQL injection, etc.
-  const { id } = req.params;
-  
-  // Look up the collection in the data store by ID
-  // When connected to a DB, replace this with a SQL query (SELECT * FROM collections WHERE id = ?)
-  const collection = collectionsStore.find(c => c.id === id);
-  
-  if (!collection) {
-    // Return 404 with standardized error format
-    return res.status(404).json({
-      code: 'NotFound',
-      description: `Collection with id '${id}' not found`,
-      id: id
+router.get('/:id', validateCollectionId, async (req, res, next) => {
+  try {
+    const { id } = req.params;
+
+    // id is already syntactically validated by validateCollectionId.
+    // For the database we use a numeric id, matching the collection.id column type.
```

Comment:
I don't know if it's necessary to mention, but the SQL-Builder is using `c.collection.id`.

```suggestion
    // For the database we use a numeric id, matching the c.collection.id column type.
```